packages/clay-toolbar/src/index.tsx

```diff
@@ -43,29 +40,24 @@ const ClayToolbar: React.FunctionComponent<IProps> & {
 } = ({
 	children,
 	className,
-	component,
 	inlineBreakpoint,
 	subnav,
 	...otherProps
 }: IProps) => {
-	const classes = classNames(
-		className,
-		'tbar',
-		{
-			'component-tbar': component,
-			'subnav-tbar': !!subnav,
-			'subnav-tbar-disabled': !!(
-				subnav &&
-				typeof subnav !== 'boolean' &&
-				subnav.disabled
-			),
-			[`tbar-inline-${inlineBreakpoint}-down`]: inlineBreakpoint,
-		},
-		subnav &&
-			typeof subnav !== 'boolean' &&
-			subnav.displayType &&
-			`subnav-tbar-${subnav.displayType}`
-	);
+	subnav =
+		typeof subnav === 'boolean'
+			? subnav === true
+				? {}
+				: undefined
+			: subnav;
```

Comment:
This is for typescript and so that our class logic doesnt have to have so many steps. By setting it to an empty object if its true, this allows us to use optional chaining below. I think this cleans it up quite a bit.

Reply:
I know that optional chaining makes things *look* neat, but I actually think it might be hiding a smell in some places (don't know about this one).

Like the similar pattern I commented on [over here](https://github.com/liferay/clay/pull/3303/files/916784ec784ae88a129392f147480a6b54062be1#r436503877), ideally TS would infer as much information as possible about what type a value has at any given point — and on that PR, we use `?.` to make up for TS's failure to infer what we would know to be true just by looking at the code. In that case, the "smell" was that we were using `?.` to say "this value might be undefined but you shouldn't blow up if it is", when the truth was "at this point we know it can't be undefined and we're just using `?.` to shut up the type checker". The alternative to using `?.` all over the place, is tweaking the code so that the inference is better and you don't need the optional chaining anymore. In that PR, you can see that assigning to a variable is enough to help TS narrow the type information. (I general like it when the type system and I agree with each other; it's not so fun when we're in conflict.)

Reply:
Thats a good point. I had another pass through that logic and think I found a better way to clean it up. Check out https://github.com/liferay/clay/pull/3325/commits/1e0fb14bd617cf13bedbc7ea936d1b39c0b6eca9#diff-60699a060cdb1fe818af4b8473b3240fR53-R60

Reply:
Looks like a nice improvement!